commit 5b5a285c7bab3851aa428a2ead14106a3f949788
Author: Greg Kroah-Hartman <gregkh@suse.de>
Date:   Tue Oct 5 11:28:57 2010 -0700

    Staging: brcm80211: bcmdefs.h: remove unused #defines
    
    There were a number of unused and unneeded defines in here, remove them.
    
    Cc: Brett Rudley <brudley@broadcom.com>
    Cc: Henry Ptasinski <henryp@broadcom.com>
    Cc: Nohee Ko <noheek@broadcom.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/staging/brcm80211/include/bcmdefs.h
+++ b/drivers/staging/brcm80211/include/bcmdefs.h
@@ -29,10 +29,6 @@
  * typedefs.h is included.
  */
 
-/* Use BCM_REFERENCE to suppress warnings about intentionally-unused function
- * arguments or local variables.
- */
-#define BCM_REFERENCE(data)	((void)data)
 
 /* Reclaiming text and data :
  * The following macros specify special linker sections that can be reclaimed
@@ -44,18 +40,10 @@
 #define bcmreclaimed 		0
 #define BCMATTACHDATA(_data)	_data
 #define BCMATTACHFN(_fn)	_fn
-#define BCMPREATTACHDATA(_data)	_data
-#define BCMPREATTACHFN(_fn)	_fn
 #define BCMINITDATA(_data)	_data
 #define BCMINITFN(_fn)		_fn
 #define BCMUNINITFN(_fn)	_fn
 #define	BCMNMIATTACHFN(_fn)	_fn
-#define	BCMNMIATTACHDATA(_data)	_data
-#define BCMOVERLAY0DATA(_sym)	_sym
-#define BCMOVERLAY0FN(_fn)	_fn
-#define BCMOVERLAY1DATA(_sym)	_sym
-#define BCMOVERLAY1FN(_fn)	_fn
-#define BCMOVERLAYERRFN(_fn)	_fn
 #ifdef mips
 #define BCMFASTPATH		__attribute__ ((__section__(".text.fastpath")))
 #else
@@ -63,14 +51,7 @@
 #endif
 
 /* Put some library data/code into ROM to reduce RAM requirements */
-#define BCMROMDATA(_data)	_data
-#define BCMROMDAT_NAME(_data)	_data
 #define BCMROMFN(_fn)		_fn
-#define BCMROMFN_NAME(_fn)	_fn
-#define BCMROMDAT_ARYSIZ(data)	ARRAYSIZE(data)
-#define BCMROMDAT_SIZEOF(data)	sizeof(data)
-#define BCMROMDAT_APATCH(data)
-#define BCMROMDAT_SPATCH(data)
 
 /* Bus types */
 #define	SI_BUS			0	/* SOC Interconnect */
@@ -224,6 +205,4 @@ typedef struct {
 /* Max. nvram variable table size */
 #define	MAXSZ_NVRAM_VARS	4096
 
-#define LOCATOR_EXTERN static
-
 #endif				/* _bcmdefs_h_ */
